src/VecSim/algorithms/hnsw/hnsw_tiered.h

```diff
@@ -275,13 +280,8 @@ HNSWIndex<DataType, DistType> *TieredHNSWIndex<DataType, DistType>::getHNSWIndex
 }
 
 template <typename DataType, typename DistType>
-void TieredHNSWIndex<DataType, DistType>::executeReadySwapJobs() {
+void TieredHNSWIndex<DataType, DistType>::executeReadySwapJobs(size_t maxJobsToRun) {
```

Comment:
why do we need it here
Also
consider moving the "best effort" implementation to also run up to a certain threshold